### PR TITLE
[feat] created separate sign in and login pages

### PR DESCRIPTION
**Pull request by saitunguturu**
### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
- built off Jinkang's work to implement separate sign up and login pages
- sign up page creates a user when provided with a valid email, then prints a successful message in console. gives an alert if there is an error
- log in page checks for valid credentials. gives an alert if the credentials are invalid or an error occurs, and prints a successful message in console otherwise

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."
**Sign Up Page**
image

Alert with invalid credentials:
image

**Log in Page**
image

Alert with invalid credentials:
image

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'
This can be reviewed all at once. For sign up, confirm that an account is created when the user enters a valid email address & password, and that an alert is shown when that is not the case. Similarly, confirm that valid credentials allow users to login, with an alert being thrown when they are invalid credentials. 

There are currently no edge cases for signing up other than having an invalid email address. (the Supabase error message is displayed for all other errors)

There are currently no edge cases for login other than invalid login credentials. (the Supabase error message is displayed for all other errors)

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."
I need to add a method to check if a user with a specific email already exists before signing up for an account. Currently, it says users have successfully signed up for an account even if you use the same email multiple times.

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'
https://supabase.com/docs/reference/javascript/auth-signup
https://supabase.com/docs/guides/auth/auth-email-passwordless
https://supabase.com/docs/guides/auth/debugging/error-codes

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"
I built off of jinkang/create-scaffold-2

CC: @ethan-tam33

### CODE REVIEW

File: app/(auth)/login/page.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+import { useState } from 'react';
+import { useRouter } from 'next/navigation';
+import supabase from '@/actions/supabase/client';
+import CustomLink from '@/components/CustomLink';
+
+export default function LoginPage() {
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  //const [isSignUp, setIsSignUp] = useState(false); // toggle between sign in / sign up, not used for now
```
Comment (jinkang-0) [inline app/(auth)/login/page.tsx:11]: We can remove the `isSignUp` state now that we're using separate pages

File: app/(auth)/sign-up/page.tsx [hunk 2]
```diff
@@ -1,10 +1,64 @@
...
+
+import { useState } from 'react';
+import { useRouter } from 'next/navigation';
+import supabase from '@/actions/supabase/client';
 import CustomLink from '@/components/CustomLink';
 
 export default function SignUpPage() {
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  //const [isSignUp, setIsSignUp] = useState(false); // toggle between sign in / sign up, not used for now
```
Comment (jinkang-0) [inline app/(auth)/sign-up/page.tsx:11]: Similar thing here, we can remove the `isSignUp` state now that we're using separate pages for login and sign up

File: app/(auth)/sign-up/page.tsx [hunk 3]
```diff
@@ -1,10 +1,64 @@
...
+      <button onClick={handleSignUp}>Sign Up</button>
     </div>
   );
+
+  // return (
+  //   <div className="flex h-full w-full flex-col items-center justify-center">
+  //     <CustomLink href="/">← Back to Home</CustomLink>
+  //     <div className="text-center">Sign up page</div>
+  //   </div>
+  // );
```
Comment (jinkang-0) [inline app/(auth)/sign-up/page.tsx:63]: We can remove these comments, the placeholder is no longer needed

File: app/(auth)/sign-up/page.tsx [hunk 4]
```diff
@@ -1,10 +1,64 @@
...
+
+    if (error) {
+      if (error.code?.includes('email_exists')) {
+        throw new Error('This email is already registered');
+      } else {
+        alert(`Error signing up user: ${error.message}`);
+      }
+    } else {
+      console.log('User signed up successfully:', data);
+    }
```
Comment (jinkang-0) [inline app/(auth)/sign-up/page.tsx:28]: nit: remove console log. Generally logging is useful on the server. However, since this is executed on the client, we should ideally have no logs for cleanliness.

File: app/(auth)/sign-up/page.tsx [hunk 5]
```diff
@@ -1,10 +1,64 @@
...
+
+  const handleSignUp = async () => {
+    const { data, error } = await supabase.auth.signUp({
+      email,
+      password,
+    });
+
+    if (error) {
+      if (error.code?.includes('email_exists')) {
+        throw new Error('This email is already registered');
```
Comment (jinkang-0) [inline app/(auth)/sign-up/page.tsx:22]: This should be an alert as well

File: app/(auth)/sign-up/page.tsx [hunk 6]
```diff
@@ -1,10 +1,64 @@
...
+      } else {
+        alert(`Error signing up user: ${error.message}`);
+      }
+    } else {
+      console.log('User signed up successfully:', data);
+    }
+
+    router.push('/');
```
Comment (jinkang-0) [inline app/(auth)/sign-up/page.tsx:30]: This `router.push` statement should be placed inside the `else` block. With its current position, the user will be redirected even if an error occurs.

File: app/(auth)/login/page.tsx [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+    });
+
+    if (error) {
+      if (error.code?.includes('invalid_credentials')) {
+        alert('Invalid login credentials. Please try again.');
+      } else {
+        alert(`Error signing in user: ${error.message}`); // in case there are other errors
+      }
+    } else {
+      console.log('User signed in successfully:', data);
```
Comment (jinkang-0) [inline app/(auth)/login/page.tsx:27]: nit: remove console log. Generally logging is useful on the server. However, since this is executed on the client, we should ideally remove this log statement.

File: app/(auth)/sign-up/page.tsx [hunk 8]
```diff
@@ -1,10 +1,64 @@
...
+  const [password, setPassword] = useState('');
+  //const [isSignUp, setIsSignUp] = useState(false); // toggle between sign in / sign up, not used for now
+  const router = useRouter();
+
+  const handleSignUp = async () => {
+    const { data, error } = await supabase.auth.signUp({
+      email,
+      password,
+    });
+
```
Comment (jinkang-0) [inline app/(auth)/sign-up/page.tsx:19]: This is optional for now, but once we have the domain configured, we should add a redirect URL to the login page (currently, clicking the confirmation link sent in the email takes the user to the home page).

**Review by jinkang-0 — COMMENTED** [review]
Left some comments for minor adjustments, but looks good overall

File: app/(auth)/login/page.tsx [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+  const router = useRouter();
+
+  const handleSignIn = async () => {
+    const { data, error } = await supabase.auth.signInWithPassword({
+      email,
+      password,
+    });
+
+    if (error) {
+      if (error.code?.includes('invalid_credentials')) {
```
Comment (jinkang-0) [inline app/(auth)/login/page.tsx:20]: nit: instead of `error.code?.includes(...)`, we should use exact match instead (e.g. `error.code && error.code === ...`)

File: app/(auth)/sign-up/page.tsx [hunk 10]
```diff
@@ -1,10 +1,56 @@
...
+    if (error) {
+      if (error.code?.includes('email_exists')) {
```
Comment (jinkang-0) [inline app/(auth)/sign-up/page.tsx:20]: similar thing here, use exact match instead of includes

**Review by jinkang-0 — COMMENTED** [review]
Thanks for addressing the changes, I just have some minor nits, but should be good to merge after
